src/App.js

```diff
@@ -98,21 +98,44 @@ function App() {
     const sortOrder = sortArr[1].toLowerCase();
 
     // Firestore query states
-    const userQuery = query(doc(db, collectionPath));
-    const [userData, userLoading, userError] = useDocumentData(userQuery);
-
-    const [subcollectionName, setSubcollectionName] = useState(null);
-    // Set subcollection to first list after user data has loaded
+    const listsQuery = query(collection(db, collectionName), where("userId", "==", userId));
+    const [listsData, listsLoading, listsError] = useCollectionData(listsQuery);
+
+    // Need listId to query tasks of that list, but want to display listName
+    const [listInfo, setListInfo] = useState({
+        listId: null,
+        listInfo: null,
+    });
+    const listNameToId = (name) => {
+        for (const list of listsData) {
+            if (list.listName === name) {
+                return list.listId;
+            }
+        }
+    }
+    // Set listId and listName to first list after lists data has loaded
     useEffect(() => {
-        if (!subcollectionName) {
-            setSubcollectionName(userData ? userData.lists[0] : null);
+        if (!listInfo.listId) {
+            setListInfo({
+                listId: listsData ? listsData[0].listId : null,
+                listName: listsData ? listsData[0].listName : null,
+            });
+        }
+    }, [listsData, listInfo.listId]);
+    const firstDifferentList = (list) => {
+        for (const list2 of listsData) {
+            if (list.listId !== list2.listId) {
+                return {
+                    listId: list2.listId,
+                    listName: list2.listName,
+                };
+            }
         }
-    }, [userData, subcollectionName]);
+    }
 
-    const itemsQuery = query(collection(db, collectionPath + subcollectionName), orderBy(sortField, sortOrder));
+    const itemsQuery = query(collection(db, `${collectionName}/${listInfo.listId}/${subcollectionName}`), orderBy(sortField, sortOrder));
```

Comment:
This:
```collection(db, ${collectionName}/${listInfo.listId}/${subcollectionName})```
is repeated a lot.  How about not repeating yourself by defining:
const tasksCollection = collection(db, ${collectionName}/${listInfo.listId}/${subcollectionName})